src/Amalgam/entity/EntityQueryBuilder.h

```diff
@@ -87,97 +102,90 @@ namespace EntityQueryBuilder
-								if(ocn_size > 1)
-									dist_params.featureParams[i].knownToUnknownDifference = EvaluableNode::ToNumber(ocn[1]);
-								if(ocn_size > 2)
-									dist_params.featureParams[i].unknownToUnknownDifference = EvaluableNode::ToNumber(ocn[2]);
-							}
-							else //treat as singular value
-							{
-								dist_params.featureParams[i].typeAttributes.nominalCount = EvaluableNode::ToNumber(en);
-							}
-						}
-						else
-						{
-							dist_params.featureParams[i].typeAttributes.nominalCount = 0.0;
-						}
+							dist_params.featureParams[i].typeAttributes.nominalCount = EvaluableNode::ToNumber(en, 1);
 						break;
 
 					case GeneralizedDistance::FDT_CONTINUOUS_NUMERIC_CYCLIC:
+						if(found && !EvaluableNode::IsNull(en))
+							dist_params.featureParams[i].typeAttributes.maxCyclicDifference = EvaluableNode::ToNumber(en);
+						else //can't be cyclic without a range
+							dist_params.featureParams[i].featureType = GeneralizedDistance::FDT_CONTINUOUS_NUMERIC;
+						break;
+
+					default:
+						break;
+					}
+				}
+			});
+
+		//get deviations
+		EvaluableNode::ConvertChildNodesAndStoreValue(deviations_node, element_names, num_elements,
+			[&dist_params](size_t i, bool found, EvaluableNode *en) {
+				if(i < dist_params.featureParams.size())
+				{
+					dist_params.featureParams[i].deviation = 0.0;
+					dist_params.featureParams[i].unknownToUnknownDistanceTerm.difference = std::numeric_limits<double>::quiet_NaN();
+					dist_params.featureParams[i].knownToUnknownDistanceTerm.difference = std::numeric_limits<double>::quiet_NaN();
+
+					//get deviations based on feature type
+					switch(dist_params.featureParams[i].featureType)
+					{
+					case GeneralizedDistance::FDT_NOMINAL_NUMERIC:
+					case GeneralizedDistance::FDT_NOMINAL_STRING:
+					case GeneralizedDistance::FDT_NOMINAL_CODE:
 						if(found && !EvaluableNode::IsNull(en))
 						{
 							if(en->EvaluableNode::IsOrderedArray())
 							{
 								auto &ocn = en->GetOrderedChildNodesReference();
 								size_t ocn_size = ocn.size();
+
 								if(ocn_size > 0)
-									dist_params.featureParams[i].typeAttributes.maxCyclicDifference = EvaluableNode::ToNumber(ocn[0]);
+									PopulateFeatureDeviationData(dist_params.featureParams[i], ocn[0]);
+
 								if(ocn_size > 1)
-									dist_params.featureParams[i].knownToUnknownDifference = EvaluableNode::ToNumber(ocn[1]);
+									dist_params.featureParams[i].knownToUnknownDistanceTerm.difference = EvaluableNode::ToNumber(ocn[1]);
+
 								if(ocn_size > 2)
-									dist_params.featureParams[i].unknownToUnknownDifference = EvaluableNode::ToNumber(ocn[2]);
+									dist_params.featureParams[i].unknownToUnknownDistanceTerm = EvaluableNode::ToNumber(ocn[2]);
 							}
 							else //treat as singular value
 							{
-								dist_params.featureParams[i].typeAttributes.maxCyclicDifference = EvaluableNode::ToNumber(en);
+								PopulateFeatureDeviationData(dist_params.featureParams[i], en);
```

Comment:
This is setting an entirely different value now. The docs were a bit hard to tease out if this is correct?

Reply:
No, I don't think so. This was just my misunderstanding of the updates. :)